indico/modules/events/contributions/controllers.py

```diff
@@ -163,10 +194,15 @@ def _process(self):
 
 class RHEditContribution(RHManageContributionBase):
     def _process(self):
-        form = ContributionForm(obj=FormDefaults(self.contrib, start_date=self.contrib.start_dt), event=self.event_new,
-                                contrib=self.contrib)
+        contrib_form_class = make_contribution_form(self.event_new)
+        custom_field_values = {'custom_{}'.format(x.contribution_field_id): x.data for x in self.contrib.field_values}
+        form = contrib_form_class(obj=FormDefaults(self.contrib, start_date=self.contrib.start_dt,
+                                                   **custom_field_values),
+                                  event=self.event_new, contrib=self.contrib)
         if form.validate_on_submit():
-            update_contribution(self.contrib, form.data)
+            field_values, custom_fields_values = self._get_field_values(form.data)
+            update_contribution(self.contrib, field_values)
+            self._update_custom_fields(self.contrib, custom_fields_values)
```

Comment:
But if I add a mixin as you said, how's the `update_contribution` going to call the methods of it?